test/sql/function/list/list_concat.test

```diff
@@ -0,0 +1,113 @@
+# name: test/sql/function/list/list_concat.test
+# description: Test list_concat function
+# group: [list]
+
+statement ok
+PRAGMA enable_verification
+
+query T
+SELECT list_concat([1, 2], [3, 4])
+----
+[1, 2, 3, 4]
+
+query T
+SELECT array_cat([1, 2], [3, 4])
+----
+[1, 2, 3, 4]
+
+query T
+SELECT list_concat(NULL, [3, 4])
+----
+NULL
+
+query T
+SELECT list_concat([1, 2], NULL)
+----
+NULL
+
+query T
+SELECT list_concat([], [])
+----
+[]
+
+query T
+SELECT list_concat([], [3, 4])
+----
+[3, 4]
+
+query T
+SELECT list_concat([1, 2], [])
+----
+[1, 2]
+
+statement error
+SELECT list_concat(['a', 'b'], [3, 4])
+
+statement error
+SELECT list_concat([1, 2], ['c', 'd'])
+
+query T
+SELECT list_concat([NULL], [NULL])
+----
+[NULL, NULL]
+
+query T
+SELECT list_concat([1, 2], [NULL])
+----
+[1, 2, NULL]
+
+query T
+SELECT list_concat([NULL], [3, 4])
+----
+[NULL, 3, 4]
+
+query T
+SELECT list_concat([[1, 2]], [[3, 4]])
+----
+[[1, 2], [3, 4]]
+
+query T
+SELECT list_concat([{a: 1}, {a: 2}], [{a: 3}, {a: 4}])
+----
+[{'a': 1}, {'a': 2}, {'a': 3}, {'a': 4}]
+
+statement ok
+CREATE TABLE test AS SELECT range % 4 i, range j, range k FROM range(16)
+
+statement ok
+CREATE TABLE lists AS SELECT i, list(j) j, list(k) k FROM test GROUP BY i
+
+query II
+SELECT i, list_concat(j, k) FROM lists
+----
+0	[0, 4, 8, 12, 0, 4, 8, 12]
+1	[1, 5, 9, 13, 1, 5, 9, 13]
+2	[2, 6, 10, 14, 2, 6, 10, 14]
+3	[3, 7, 11, 15, 3, 7, 11, 15]
+
+statement error
+SELECT i, list_concat(j, cast(k AS VARCHAR)) FROM lists
+
+# list_append(l, e) is an alias for list_concat(l, list_value(e))
+query T
+SELECT list_append([1, 2], 3)
+----
+[1, 2, 3]
+
+query T
+SELECT list_append([1, 2], NULL)
+----
+[1, 2, NULL]
+
+query T
+SELECT list_append(NULL, 3)
+----
+NULL
+
+query II
+SELECT i, list_append(list_concat(j, k), i) FROM lists
+----
+0	[0, 4, 8, 12, 0, 4, 8, 12, 0]
```

Comment:
Could we add some tests with lists of different types (e.g. `list_concat([1::tinyint], [2::int])`, or more subtly `list_concat([0.1], [10])`